components/nimbus/src/lib.rs

```diff
@@ -36,22 +37,24 @@ use once_cell::sync::OnceCell;
 use persistence::{Database, StoreId, Writer};
 use serde_derive::*;
 use serde_json::{Map, Value};
-use std::collections::HashSet;
 use std::path::PathBuf;
 use std::sync::Mutex;
+use std::{collections::HashSet, str::FromStr};
 use updating::{read_and_remove_pending_experiments, write_pending_experiments};
 use uuid::Uuid;
 
 const DEFAULT_TOTAL_BUCKETS: u32 = 10000;
 const DB_KEY_NIMBUS_ID: &str = "nimbus-id";
-
+pub const DB_KEY_INSTALLATION_DATE: &str = "installation-date";
 // The main `NimbusClient` struct must not expose any methods that make an `&mut self`,
 // in order to be compatible with the uniffi's requirements on objects. This is a helper
 // struct to contain the bits that do actually need to be mutable, so they can be
 // protected by a Mutex.
 #[derive(Default)]
 struct InternalMutableState {
     available_randomization_units: AvailableRandomizationUnits,
+    installation_date: Option<DateTime<Utc>>,
```

Comment:
Note to self, we don't need this here, the caller can pass it as a part of the app context

Reply:
Ahh, nevermind my other comment, it seems you may be thinking along the same lines.